charts/opentelemetry-operator/release/release.go

```diff
@@ -0,0 +1,144 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"reflect"
+	"strings"
+
+	"gopkg.in/yaml.v3"
+)
+
+const (
+	operatorChartPath = ".."
+	appVersion        = "appVersion"
+)
+
+// getTemplates renders all the template files in this Helm chart and stores them into a map whose keys are the
+// Kubernetes resource kind plus name and values are the Kubernetes resource itself.
+func getTemplates() (map[string]K8sObject, error) {
+	templates := make(map[string]K8sObject)
+
+	// Use `helm template` command to render all the template files.
+	cmd := exec.Command("helm", "template", operatorChartPath)
+	out, err := cmd.Output()
+	if err != nil {
+		return nil, err
+	}
+
+	// Process all the YAML files and store the key-value pairs into templates map.
+	rawYAMLFiles := strings.Split(string(out), "---\n")
+	for _, rawYAMLFile := range rawYAMLFiles {
+		var curObject K8sObject
+		err = yaml.Unmarshal([]byte(rawYAMLFile), &curObject)
+		if err != nil {
+			return nil, err
+		}
+		templates[fmt.Sprintf("%s#%v", curObject.Kind, curObject.Metadata["name"])] = curObject
+	}
+
+	return templates, nil
+}
+
+// updateCRD writes the latest OpenTelemetry Collector CRD template file to the Helm chart's Collector CRD file.
+func updateCRD(object K8sObject, collectorCRDPath string) error {
+	out, err := yaml.Marshal(object)
+	if err != nil {
+		return err
+	}
+
+	err = os.WriteFile(collectorCRDPath, out, 0644)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// updateImageTags retrieve the latest image tags and update the values.yaml and Chart.yaml respectively.
+func updateImageTags(object K8sObject, valuesYAMLPath string, chartYAMLPath string) error {
+	// Retrieve the latest image repository and tag of the two container images.
+	managerImage := strings.Split(object.Spec["template"].(map[string]interface{})["spec"].(map[string]interface{})["containers"].([]interface{})[0].(map[string]interface{})["image"].(string), ":")
```

Comment:
It's because Operator contains two images: one is `manager`, the other is `kube-rbac-proxy`. Also, `manager` is the container's name. So I think `managerImage` is more informative.

Reply:
ok, makes sense